lib/lab_manager/provider/vsphere.rb

```diff
@@ -105,7 +108,20 @@ def create_vm(opts = {})
     end
 
     def terminate_vm(_opts = {})
-      VSphere.connect.with do |_vs|
+      fail ArgumentError, 'Virtual machine data not present' unless instance_uuid
+
+      VSphere.connect.with do |vs|
+        Retryable.retryable(tries: 3) do
+          unless vs.get_virtual_machine(instance_uuid)['power_state'] == 'poweredOff'
+            vs.vm_power_off('force' => true, 'instance_uuid' => instance_uuid)
+          end
```

Comment:
btw, this power_off implementation will be probably as as in `power_off` method.